solidity/test/relay/LightRelayMaintainerProxy.test.ts

```diff
@@ -0,0 +1,532 @@
+        })
+
+        it("should emit the MaintainerAuthorized event", async () => {
+          await expect(tx)
+            .to.emit(lightRelayMaintainerProxy, "MaintainerAuthorized")
+            .withArgs(maintainer.address)
+        })
+      })
+    })
+  })
+
+  describe("deauthorize", () => {
+    context("When called by non-owner", () => {
+      it("should revert", async () => {
+        await expect(
+          lightRelayMaintainerProxy
+            .connect(thirdParty)
+            .deauthorize(maintainer.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called by the owner", () => {
+      context("when the maintainer is not authorized", () => {
+        before(async () => {
+          await createSnapshot()
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            lightRelayMaintainerProxy
+              .connect(governance)
+              .deauthorize(maintainer.address)
+          ).to.be.revertedWith("Maintainer is not authorized")
+        })
+      })
+
+      context("when the maintainer is authorized", () => {
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          // Authorize the maintainer first
+          await lightRelayMaintainerProxy
+            .connect(governance)
+            .authorize(maintainer.address)
+
+          tx = await lightRelayMaintainerProxy
+            .connect(governance)
+            .deauthorize(maintainer.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should deauthorize the address", async () => {
+          expect(
+            await lightRelayMaintainerProxy.isAuthorized(maintainer.address)
+          ).to.be.false
+        })
+
+        it("should emit the MaintainerDeauthorized event", async () => {
+          await expect(tx)
+            .to.emit(lightRelayMaintainerProxy, "MaintainerDeauthorized")
+            .withArgs(maintainer.address)
+        })
+      })
+    })
+  })
+
+  describe("updateLightRelay", () => {
+    context("When called by non-owner", () => {
+      it("should revert", async () => {
+        await expect(
+          lightRelayMaintainerProxy
+            .connect(thirdParty)
+            .updateLightRelay(thirdParty.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called by the owner", () => {
+      context("when called with zero address", () => {
+        it("should revert", async () => {
+          await expect(
+            lightRelayMaintainerProxy
+              .connect(governance)
+              .updateLightRelay(ZERO_ADDRESS)
+          ).to.be.revertedWith("New light relay must not be zero address")
+        })
+      })
+
+      context("When called with a non-zero address", () => {
```

Comment:
```suggestion
      context("when called with a non-zero address", () => {
```

Reply:
Done in 462925b1bd07e05f87beb7c9fcba541d335c0be1.